### DISCUSSION TITLE
The difference between `dev` and `master` branch

### DISCUSSION THREAD

**Discussion by xudong963** [qdrant/qdrant#1274] [Q&A] [answered]
When I read https://github.com/qdrant/qdrant/blob/master/CONTRIBUTING.md, I noticed `Fork the repo and create your branch from dev.`

However, when I looked through pull requests, I found some PRs merged to dev and some merged to master.

So I want to ask what's the difference between `dev` and `master` branch ?

**Comment by xudong963** [comment]
Btw, seems Qdrant doesn't support macos as development environment?

**Reply by generall** [reply]
Do you have problems running qdrant on macos? Could you please report errors you are facing?

**Reply by xudong963** [reply]
When I run `Cargo build`, I got the following error:

```rust
   Compiling zstd-safe v5.0.2+zstd.1.5.2
   Compiling rayon-core v1.10.1
   Compiling geo-types v0.7.7
   Compiling crossbeam-deque v0.8.2
error: expected `,`, found `.`
   --> /Users/xudong/.cargo/registry/src/github.com-1ecc6299db9ec823/geo-types-0.7.7/src/geometry/geometry_collection.rs:111:25
    |
111 | #[deprecated(since = 0.7.5, note = "Use `GeometryCollection::from(vec![geom])` instead.")]
    |                         ^ expected `,`

   Compiling webpki v0.22.0
error: could not compile `geo-types` due to previous error
warning: build failed, waiting for other jobs to finish...
➜  qdrant git:(master) cargo -V
cargo 1.67.0-nightly (f6e737b1e 2022-12-02)
➜  qdrant git:(master) rustc -V
rustc 1.67.0-nightly (e1d819583 2022-12-05)
➜  qdrant git:(master)
```

**Reply by xudong963** [reply]
Hi @generall , do you have any thought?

**Reply by generall** [reply]
We usually build it with stable rust: `1.65.0`

**Comment by generall** [comment, marked as answer]
Hi @xudong963! We are trying to follow git pattern, where we develop new features in dev and then merge them into master on releases:
![image](https://user-images.githubusercontent.com/1935623/206854787-f76e7385-dc0e-4bbb-ac3b-a98f880f6eb2.png)

In general, we are trying to maintain the following guarantees:

- master should be backward compatible with the latest release and clients, it should pass all tests
- dev should only pass all tests, backward compatibility can be delayed to the moment of merge
